src/lib/ui/Button.tsx

```diff
@@ -0,0 +1,23 @@
+import classNames from 'classnames'
+
+interface ButtonProps {
+  children?: string
+  variant?: 'solid' | 'outline' | 'neutral'
+  size?: 'small' | 'medium' | 'large'
+  disabled?: boolean
+}
+
+export function Button(props: ButtonProps) {
+  const { variant = 'outline', children, size = 'medium', disabled } = props
```

Comment:
I feel like for defaults like this we should prefer

```ts
export function Button({ variant = 'outline', children, size = 'medium', disabled }: ButtonProps) {
```

Reply:
I usually destructure in a separate line in case we need to forward all the props to a child component. Doing it inline would be a bit more typing:

```tsx
export function Button({ variant = 'outline', children, size = 'medium', disabled, ...otherProps }: ButtonProps) {
   return <SpecialButton  variant={variant} children={children} size={size} disabled={disabled} {...otherProps} />
}
```

Reply:
Yea, I have been frustrated by wanting to use these two incompatible patterns at the same time. In this case though we are not forwarding. Also with default values, we can't forward everything since we need to assign the defaults.

My main thought here was having the defaults in the function signature might make for better TS doc magic, but that was just a theory.

Reply:
>  Also with default values, we can't forward everything since we need to assign the defaults.

Child components would be able to specify their own defaults.

> Yea, I have been frustrated by wanting to use these two incompatible patterns at the same time.

Yep. As long as we're cool with having different patterns when needed I think we can destructure it in function signature in this case.

Reply:
Updated in https://github.com/seamapi/react/pull/117/commits